tests/e2e/e2e_health_probe_test.go

```diff
@@ -253,5 +262,27 @@ server {
 			// Expect it to be up and running in it's receiver namespace
 			Expect(Td.WaitForPodsRunningReady(ns, 60*time.Second, len(pods), nil)).To(Succeed())
 		})
+
+		It("Incorrectly configures Pods' TCPSocket probes so they fail as expected", func() {
+			const ns = "healthprobe"
+			// Install OSM
+			Expect(Td.InstallOSM(Td.GetOSMInstallOpts())).To(Succeed())
+
+			Expect(Td.CreateNs(ns, nil)).To(Succeed())
+			Expect(Td.AddNsToMesh(true, ns)).To(Succeed())
+
+			pod := makePod("nginx-tcp", incorrectTCP)
+
+			_, err := Td.Client.CoreV1().Secrets(ns).Create(context.TODO(), tls, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			_, err = Td.Client.CoreV1().ConfigMaps(ns).Create(context.TODO(), conf, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			_, err = Td.Client.CoreV1().Pods(ns).Create(context.TODO(), pod, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Expect it to not be ready
+			Expect(Td.WaitForPodsRunningReady(ns, 60*time.Second, 1, nil)).To(MatchError("not all pods were Running & Ready in NS healthprobe after 1m0s"))
```

Comment:
Nit: define the timeout once and use it to parameterize the expected error message. Or omit the duration from the message if the expected error doesn't have to match the full error.